OOP/Restaurant/item.py

```diff
@@ -0,0 +1,164 @@
+"""Module Contains classes Item, OrderedITem, Stock, Order, Payment  and Delivery from a 
+restaurant classes representation"""
+
+from restaurant import Customer
+from employee import Employee
+
+class Item:
+    """Class representing the single item from menu"""
+    def __init__(self, category,item_name, price) -> None:
+        self.item_name = item_name
+        self.price=price
+        self.category=category
+        self.item_dic = None
+
+    def add_item(self, menu):
+        """Adds an item to menu"""
+        self.item_dic ={self.category:{self.item_name:self.price}}
+        menu.add_menu(self.item_dic)
+
+class OrderedItem(Item):
+    """Class representing an Item ordered by a customer"""
+
+    def __init__(self,date: str ,item: Item, quantity: int, customer: Customer) -> None:
+        super().__init__(item.category, item.item_name, item.price)
+        self.date=date
+        self.quantity = quantity
+        self.customer= customer
+        self.total=None
+
+    def calculate_total(self):
+        """Returns the product from the quantity and price for the ordered item"""
+        return self.price * self.quantity
+
+    def show_details(self):
+        """Show the details for the ordered item"""
+        self.total= self.calculate_total()
+        print("Ordered Item")
+        print(f'Date: {self.date}\n'
+              f'Ordered Item: {self.item_name}\n'
+              f'Item Price: {self.price} \n'
+              f'Quantity: {self.quantity} \n'
+              f'Total Amount: {self.total} \n')
+
+    def __str__(self):
+        return f"Item: {self.item_name}, Category: {self.category} "
+
+    def __repr__(self):
+        return self.__str__()
+
+
+class Stock:
+    """Class representing the available stock from a restaurant
+    It contains the quantities for each item on the menu"""
+    def __init__(self) -> None:
+        self.inventory={}
+
+    def add_stock(self,item_name, quantity):
+        """Add an item or increment its quantity"""
+
+        if item_name in self.inventory:
+            self.inventory[item_name] += 1
+        else:
+            self.inventory[item_name]=quantity
+
+
+    def deduct_item(self, item: Item):
+        """Rest the stock quantity from a ordered item"""
+        item_name=item.item_name
+        quantity=item.quantity
```

Comment:
why you are moving this values to separate variables instead of just using `item.item_name` `item.quantity` directly, no need for separate variable

Reply:
`quantity` is attribute of `OrderItem` class, not `Item` class 
![image](https://github.com/user-attachments/assets/85fb693c-cd12-41ac-80b9-b38c239c1630)